cmd/fdsn-quake-consumer/sc3ml.go

```diff
@@ -203,10 +204,13 @@ func (e *event) save() error {
 
 	_, err = txn.Exec(`DELETE FROM fdsn.event WHERE PublicID = $1 AND ModificationTime <= $2`, e.PublicID, e.ModificationTime)
 	if err != nil {
-		txn.Rollback()
+		if e := txn.Rollback(); e != nil {
+			log.Printf("Rollback Failed: %v", e)
+		}
 		return err
 	}
 
+	//nolint:gosec	// Unavoidable string concatenation
 	_, err = txn.Exec(`INSERT INTO fdsn.event(`+strings.TrimSuffix(insert, ", ")+`) VALUES( `+strings.TrimSuffix(params, ", ")+` )`, values...)
```

Comment:
Would a `fmt.Sprintf` trigger a linter warning here as well?

I understand that this is safe, but it concerns me that we have a //nolint:gosec next to an `sql` function.

Could you try creating the string on another line (maybe with `fmt.Sprintf` for better readability) and passing it in as a variable?

Reply:
Tried with fmt.Sprintf still got the warning of 
```cmd/fdsn-quake-consumer/sc3ml.go:213: G201: SQL string formatting (gosec)```

Reply:
What if you assign the string to a variable and pass it into `txn.Exec` like that?

Reply:
Yes that's what I did.
```
	q := fmt.Sprintf("INSERT INTO fdsn.event(%s) VALUES(%s)", strings.TrimSuffix(insert, ", "), strings.TrimSuffix(params, ", "))
	_, err = txn.Exec(q, values...)
```
The code above still being rejected.
But the error is at the line of fmt . 
I think the linter checks context to see whether a string formatting is for SQL execution.

Similar static analysis and workarounds happened here:
https://github.com/GeoNet/haz/pull/220/files#diff-c32caf3c852145c15537fee84e4b625aR57

Reply:
I'll expand the "INSERT INTO" to resolve the lint error.

Reply:
@danieldooley I've updated the PR. Could you please check again? Thanks.